source/m2mconnectionhandlerpimpl.cpp

```diff
@@ -87,526 +175,589 @@ bool M2MConnectionHandlerPimpl::bind_connection(const uint16_t listen_port)
-                            _received_packet_address->_stack = M2MInterface::LwIP_IPv4;
-                            _received_packet_address->_length = 4;
-                            _sa_dst.sin_family = AF_INET;
-                            _sa_dst.sin_port = ntohs(server_port);
-                            memcpy(&_sa_dst.sin_addr, _received_packet_address->_address, _received_packet_address->_length);
-                            if (connect(_socket_server, (const struct sockaddr *)&_sa_dst, _slen_sa_dst) != 0) {
-                                success = false;
-                                tr_error("M2MConnectionHandlerPimpl::resolve_hostname - failed to connect %s, %s", ip_address, strerror(errno));
-                            } else {
-                                success = true;
-                                tr_debug("M2MConnectionHandlerPimpl::resolve_hostname - connected to %s\n", ip_address);
-                            }
-                        }
-                        break;
-                    case AF_INET6:
-                        a6 =  (struct sockaddr_in6*)addr_info->ai_addr;
-                        if(a6) {
-                            inet_ntop(AF_INET6,
-                                      &(a6->sin6_addr),
-                                      ip_address,
-                                      INET6_ADDRSTRLEN);
-
-                            inet_pton(AF_INET6, ip_address, &_resolved_address);
-                            _received_packet_address->_port = server_port;
-                            memcpy(_received_packet_address->_address,
-                                   _resolved_address,
-                                   sizeof(_resolved_address));
-                            _received_packet_address->_stack = M2MInterface::LwIP_IPv6;
-                            _received_packet_address->_length = sizeof(_resolved_address);
-
-                            _sa_dst6.sin6_family = AF_INET6;
-                            _sa_dst6.sin6_port = ntohs(server_port);
-                            memcpy(&_sa_dst6.sin6_addr,
-                                   _received_packet_address->_address,
-                                   _received_packet_address->_length);
-                            _sa_dst6.sin6_scope_id = 0;
-                            if (connect(_socket_server, (const struct sockaddr *)&_sa_dst6, _slen_sa_dst6) != 0) {
-                                success = false;
-                                tr_error("M2MConnectionHandlerPimpl::resolve_hostname - failed to connect %s\n", ip_address);
-                            } else {
-                                success = true;
-                                tr_debug("M2MConnectionHandlerPimpl::resolve_hostname - connected to %s\n", ip_address);
-                            }
-                        }
-                        break;
-                }
```

Comment:
This method might be const as it is not supposed to be modifying the object instance.